core/lib/zksync_core/src/state_keeper/seal_criteria/conditional_sealer.rs

```diff
@@ -114,3 +142,31 @@ impl ConditionalSealer {
         ]
     }
 }
+
+/// Implementation of [`ConditionalSealer`] that never seals the batch.
+/// Can be used in contexts where, for example, state keeper configuration is not available,
+/// or the decision to seal batch is taken by some other component.
+#[derive(Debug)]
```

Comment:
what are the examples of such contexts? EN is mentioned elsewhere - makes sense that the EN doesn't seal l1 batches (for now). But I think the reason for this is that we are afraid of divergencies - in theory the EN should apply the same sealing criteria and panic if the l1 batch is sealed "incorrectly".

Reply:
> what are the examples of such contexts? 

API server on EN that accepts any transaction and defers checks to the main node.

> But I think the reason for this is that we are afraid of divergencies

We're not afraid of divergencies. The sealer on the main node uses `StateKeeperConfig` and EN doesn't currently have a mechanism to synchronize the settings with the main node if they ever change (which absolutely can happen).

Reply:
Uh, and also unit tests, obviously.